dind/dind.sh

```diff
@@ -32,11 +32,15 @@ cmd="dockerd \
 
 # nanny and restart on crashes
 until eval $cmd; do
-  echo "Docker crashed with exit code $?.  Respawning.." >&2
-  # if we just restart it won't work, so start it (it wedges up) and
-  # then kill the wedgie and restart it again and ta da... yea, seriously
-  pidfile=/var/run/docker/libcontainerd/docker-containerd.pid
-  kill -9 $(cat $pidfile)
-  rm $pidfile
-  sleep 1
+  echo "Docker crashed with exit code $?. Trying to launch with overlay2.override_kernel_check option.." >&2
+  cmd="$cmd --storage-opt overlay2.override_kernel_check=1"
+  until eval $cmd; do
+    echo "Docker crashed with exit code $?.  Respawning.." >&2
+    # if we just restart it won't work, so start it (it wedges up) and
+    # then kill the wedgie and restart it again and ta da... yea, seriously
+    pidfile=/var/run/docker/libcontainerd/docker-containerd.pid
+    kill -9 $(cat $pidfile)
+    rm $pidfile
+    sleep 1
+  done
```

Comment:
For example, i changed `dockerd` to `dockerd1` to check how it will exit if cmd will fail
```
/usr/local/bin # ./dind.sh
+ grep -Eh -w -m1 overlay|aufs /proc/filesystems
+ cut -f2
+ fsdriver=overlay
+ [ overlay == overlay ]
+ fsdriver=overlay2
+ cmd=dockerd1          --host=unix:///var/run/docker.sock              --host=tcp://0.0.0.0:2375               --storage-driver=overlay2
+ eval dockerd1 --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 --storage-driver=overlay2
+ dockerd1 --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 --storage-driver=overlay2
./dind.sh: eval: line 1: dockerd1: not found
+ echo Docker crashed with exit code 127. Trying to launch with overlay2.override_kernel_check option..
Docker crashed with exit code 127. Trying to launch with overlay2.override_kernel_check option..
+ cmd=dockerd1          --host=unix:///var/run/docker.sock              --host=tcp://0.0.0.0:2375               --storage-driver=overlay2 --storage-opt overla
y2.override_kernel_check=1
+ eval dockerd1 --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 --storage-driver=overlay2 --storage-opt overlay2.override_kernel_check=1
+ dockerd1 --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 --storage-driver=overlay2 --storage-opt overlay2.override_kernel_check=1
./dind.sh: eval: line 1: dockerd1: not found
+ echo Docker crashed with exit code 127.  Respawning..
Docker crashed with exit code 127.  Respawning..
+ pidfile=/var/run/docker/libcontainerd/docker-containerd.pid
+ cat /var/run/docker/libcontainerd/docker-containerd.pid
cat: can't open '/var/run/docker/libcontainerd/docker-containerd.pid': No such file or directory
+ kill -9
sh: you need to specify whom to kill
```